dev/test-dependencies.sh

```diff
@@ -47,7 +47,7 @@ OLD_VERSION=$($MVN -q \
     -Dexec.executable="echo" \
     -Dexec.args='${project.version}' \
     --non-recursive \
-    org.codehaus.mojo:exec-maven-plugin:1.6.0:exec)
+    org.codehaus.mojo:exec-maven-plugin:1.6.0:exec | grep -e '[0-9]\.[0-9]\.[0-9]')
```

Comment:
I used `[0-9]` because `\d` is only accepted on Mac OS.

Reply:
This would still match `3.1.0-SNAPSHOT` wouldn't it - was the idea not to match it?
You could exclude "SNAPSHOT" matches but that might be just a band-aid.

You might need `[0-9]+` in each case to account for double-digit versions.

`\d` might work with `-E` but not worth it.

Reply:
Thanks . Yes. It matches SNAPSHOT, too.

Reply:
This is only grepping lines with version like string.

Reply:
Oh, just to be clear, what are you trying to match or not match here?

Reply:
The first possible double-digit might be 2.4.10, but it's unlikely since the current one will be 2.4.6.

Reply:
Sorry for not being clear. In the script, `OLD_VERSION` became `Falling back to archive.apache.org to download Maven 3.1.0-SNAPSHOT` instead of `3.1.0-SNAPSHOT` if `build/mvn` did fallback. 

Then, `pom.xml` file is corrupted like the following at the end and the exit code become `1` instead of `0`.
```
-    <version>3.1.0-SNAPSHOT</version>
+    <version>Falling</version>
```

This PR will remove the  `Falling back to archive.apache.org to download Maven` .

Reply:
I'll update the PR description, too.

Reply:
Maybe I'm still missing it, but this will still match the whole line - do you need `-o` on the grep to just return the matching part? I'd still account for double-digit versions here to avoid errors one day when there is a .10 or whatever.

Reply:
It's two lines, @srowen .

**NO FALLBACK**
```
$ build/mvn -q -Dexec.executable="echo" -Dexec.args='${project.version}' --non-recursive org.codehaus.mojo:exec-maven-plugin:1.6.0:exec
Using `mvn` from path: /Users/dongjoon/APACHE/spark-merge/build/apache-maven-3.6.3/bin/mvn
3.1.0-SNAPSHOT
```

**FALLBACK**
```
$ build/mvn -q -Dexec.executable="echo" -Dexec.args='${project.version}' --non-recursive org.codehaus.mojo:exec-maven-plugin:1.6.0:exec
Falling back to archive.apache.org to download Maven
Using `mvn` from path: /Users/dongjoon/APACHE/spark-merge/build/apache-maven-3.6.3/bin/mvn
3.1.0-SNAPSHOT
```

**In the script**
```
$ echo (build/mvn -q -Dexec.executable="echo" -Dexec.args='${project.version}' --non-recursive org.codehaus.mojo:exec-maven-plugin:1.6.0:exec)
Using `mvn` from path: /Users/dongjoon/APACHE/spark-merge/build/apache-maven-3.6.3/bin/mvn
Falling back to archive.apache.org to download Maven 3.1.0-SNAPSHOT
```

Reply:
In the script, `$(...)` merges two lines into one.

Reply:
BTW, is the following working in `grep` on Mac?
> You might need [0-9]+ in each case to account for double-digit versions.

Reply:
It works but you need `-E` as that is 'extended' regex syntax, like many things. I think `-E` is standard across GNU / BSD grep.

OK if you tell me this works, then I'm missing something, but it seems like this grep does not exclude the line I think you are trying to exclude, unless you want to print only the matching version from that line, and that's `-o` I understand it's picking up "Falling" so I am assuming something needs to ignore this line or only extract the version. `grep -v Falling` would just ignore that line

Reply:
Thanks. BTW, `grep -v Falling` is not robust enough. If someone adds another warning, this situation will repeat again.

Reply:
I'll update to use `grep -E '[0-9]+\.[0-9]+\.[0-9]+'`.